ReactAndroid/src/main/java/com/facebook/react/uimanager/TouchTargetHelper.java

```diff
@@ -144,8 +148,19 @@ private static boolean isTransformedTouchPointInView(
       localX = localXY[0];
       localY = localXY[1];
     }
-    if ((localX >= 0 && localX < (child.getRight() - child.getLeft()))
-        && (localY >= 0 && localY < (child.getBottom() - child.getTop()))) {
+    float hitSlopTop = 0;
```

Comment:
I don't think it's sufficient to do it this way. If you take a look at the method above (`findTouchTargetView`) you'll see that touch target is determined by traversing view hierarchy based on some order (from last children to first) and there is a stop condition that ends the process whenever we hit a child view which passes the test from this method (`isTransformedTouchPointInView`).

If I understand correctly you expect "hitSlop" to "extend" area of the View. So taking the above into account there is a chance that touch will be attributed to a neighbour view that intersects with the "hitSlop" area of the view we expect to receive the touch (see [1]). In even more complicated case we may have a nested view hierarchy as a neighbour that has "hitSlop" set for some children deep down the hierarchy and in this case we won't event traverse that hierarchy as the root of it would return `false` from this test method (see [2]).

The main issue here is that in RN native views doesn't really now if they have touch handlers registered or not in JS. But I think that the whole point of having this property is to support the aforementioned cases too and we should somehow prfioritize views with "hitSlop" set, no?

1)
![](http://ibin.co/2VmkkVo9RaJ3)

2)
![](http://ibin.co/2Vml9B3HJzTK)